pydecred/rpc.py

```diff
@@ -363,30 +378,40 @@ def ticketVWAP(self, start=None, end=None):
         Returns:
             float: The volume weighted average price.
         """
-        return self.call("ticketvwap", start, end)
+        return self.call(
+            "ticketvwap", *([start] if start else []), *([end] if start and end else [])
+        )
 
-    def txFeeInfo(self, blocks=None, rangeStart=None, rangeEnd=None):
+    def txFeeInfo(self, blocks=0, rangeStart=0, rangeEnd=0):
         """
         Get various information about regular transaction fees from the mempool,
         blocks, and difficulty windows.
 
         Args:
-            blocks (int): Optional. Default=None. The number of blocks to calculate transaction fees
-                for, starting from the end of the tip moving backwards.
-            rangeStart (int): Optional. Default=None. The start height of the block range to calculate
-                transaction fees for.
-            rangeEnd (int): Optional. Default=None. The end height of the block range to calculate
-                transaction fees for.
+            blocks (int): Optional. Default=0. The number of blocks to calculate
+                transaction fees for, starting from the end of the tip moving backwards.
+            rangeStart (int): Optional. Default=0. The start height of the block
+                range to calculate transaction fees for.
+            rangeEnd (int): Optional. Default=0. The end height of the block
+                range to calculate transaction fees for.
 
         Returns:
-            dict[str]FeeInfoResult: FeeInfoResults for the keys
-                "feeinfomempool", "feeinfoblocks", and "feeinforange".
```

Comment:
Didn't you have a type that mirrored `TicketFeeInfoResult` and `TxFeeInfoResult` before? If so, that was better. If not, you should do that.

Reply:
Yes, FeeInfoResult is covers all of them, if I follow correctly. But feeinfomempool holds one FeeInfoResult and feeinfoblocks holds an array of FeeInfoResult. Do you mean have a parent type, and not use a dict to store them?

Reply:
I'm not sure we're on the same page. Your `FeeInfoResult` covers all of the `FeeInfo*` types in dcrd, but dcrd `rpcclient.Client`'s `TicketFeeInfo` and `TxFeeInfo` methods return the `TicketFeeInfoResult` and `TxFeeInfoResult` types, respectively. You are returning a Python `dict` instead of mirroring those types. 

```
type TicketFeeInfoResult struct {
	FeeInfoMempool FeeInfoMempool  `json:"feeinfomempool"`
	FeeInfoBlocks  []FeeInfoBlock  `json:"feeinfoblocks"`
	FeeInfoWindows []FeeInfoWindow `json:"feeinfowindows"`
}

type TxFeeInfoResult struct {
	FeeInfoMempool FeeInfoMempool `json:"feeinfomempool"`
	FeeInfoBlocks  []FeeInfoBlock `json:"feeinfoblocks"`
	FeeInfoRange   FeeInfoRange   `json:"feeinforange"`
}


```

Reply:
Ok, I see it now. Sorry about that.